packages/runtime/runtime-utils/src/fluidObjectReferencesUtils.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IFluidObjectReferences } from "@fluidframework/runtime-definitions";
+
+/**
+ * Helper function that normalizes the path of the given Fluid object references and prefixs them with the given id.
+ * @param references - The list of Fluid object references whose paths are to be normalized and prefixed.
+ * @param prefixId - The id to prefix to the Fluid object references path.
+ */
+export function normalizeAndPrefixReferencesPath(references: IFluidObjectReferences[], prefixId: string) {
+    for (const reference of references) {
+        let normalizedPath = reference.path;
+        while (normalizedPath.startsWith("/")) {
+            normalizedPath = normalizedPath.substr(1);
+        }
+        while (normalizedPath.endsWith("/")) {
+            normalizedPath = normalizedPath.substr(0, normalizedPath.length - 1);
+        }
+        reference.path = `${prefixId}/${normalizedPath}`;
+    }
+}
+
+/**
+ * Helper function that clones the given list of Fluid object references.
+ * @param references - The list of Fluid object references to be cloned.
+ * @returns - A clone of the given Fluid object references.
+ */
+export function cloneFluidObjectReferences(references: IFluidObjectReferences[]): IFluidObjectReferences[] {
```

Comment:
does not look specialization is needed here. Can we use deepClone()?
I do not remember where we stand re using lodash - I think Curtis does not like it, but we use it :)

Reply:
I did use deepClone() at some point in time. But it gave me an error and I had to disable an eslint warning. So, that got me thinking that we shouldn't be using it unless necessary. The clone needed here seemed straightforward so I just implemented it.

Reply:
I spoke to Curtis and it seems like loadash increases the package size a lot. We may plan to remove it at some point. I will keep this implementation so that I don't add more dependencies.